code/modules/reagents/chemistry/recipes/drugs.dm

```diff
@@ -24,10 +24,22 @@
 /datum/chemical_reaction/methamphetamine
 	name = /datum/reagent/drug/methamphetamine
 	id = /datum/reagent/drug/methamphetamine
-	results = list(/datum/reagent/drug/methamphetamine = 4)
-	required_reagents = list(/datum/reagent/medicine/ephedrine = 1, /datum/reagent/iodine = 1, /datum/reagent/phosphorus = 1, /datum/reagent/hydrogen = 1)
+	results = list(/datum/reagent/drug/methamphetamine = 5)
+	required_reagents = list(/datum/reagent/medicine/ephedrine = 1, /datum/reagent/iodine = 1, /datum/reagent/phosphorus = 1, /datum/reagent/hydrogen = 1, /datum/reagent/toxin/acid/fluacid = 1)
 	required_temp = 374
 
+/datum/chemical_reaction/methamphetamine/on_reaction(datum/reagents/holder)
+	var/datum/reagents/R = new/datum/reagents(5)
+	R.my_atom = src
+	R.add_reagent(/datum/reagent/toxin/acid, 5)
+	var/smoke_radius = round(0.5, 0.5)
+	var/location = get_turf(holder.my_atom)
+	var/datum/effect_system/smoke_spread/chem/S = new
+	playsound(location, 'sound/effects/smoke.ogg', 50, 1, -3)
+	if(S)
+		S.set_up(R, smoke_radius, location, 0)
+		S.start()
```

Comment:
```suggestion
/datum/chemical_reaction/methamphetamine/on_reaction(datum/reagents/holder)
	var/datum/reagents/R = new/datum/reagents(5)
	R.my_atom = holder.my_atom
	R.add_reagent(/datum/reagent/toxin/acid/fluacid, 5)
	var/datum/effect_system/smoke_spread/chem/S = new
	S.set_up(R, 0.5, holder.my_atom, 0)
	S.start()
	
	playsound(location, 'sound/effects/smoke.ogg', 50, 1, -3)
```
Couple of fixes to make the code look prettier and also somewhat more operational

Reply:
Added the requested changes